cadence/src/client.rs

```diff
@@ -686,6 +690,28 @@ impl StatsdClientBuilder {
         self
     }
 
+    /// Add a default tag with key and value to every metric published by the
+    /// built [StatsdClient].
+    pub fn with_tag<K, V>(mut self, key: K, value: V) -> Self
+    where
+        K: AsRef<str>,
+        V: AsRef<str>,
+    {
+        self.tags
+            .push((Some(key.as_ref().to_string()), value.as_ref().to_string()));
+        self
+    }
```

Comment:
I think it will be impossible to avoid using `String`. If so, `ToString` is the right trait to use, rather than `AsRef<str>`.